### PR TITLE
Add RandomNumberGenerator.GetBytes(Span) overload

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/22690
cc: @bartonjs

### CODE REVIEW

File: src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            Assert.NotEqual(first, second);
+        }
+
+        [Fact]
+        public static void GetBytes_Span_ZeroCount()
+        {
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                var rand = new byte[1] { 1 };
+                rng.GetBytes(new Span<byte>(rand, 0, 0));
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs]: Are casting to span (above) and calling `new Span` (here) the correct style?  Or should we be using an AsSpan / Slice method? (Not that I know they exist, just going by terms from various meetings)

File: src/System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj [hunk 2]
```diff
@@ -117,6 +117,7 @@
     <Compile Include="HmacTests.cs" />
     <Compile Include="IncrementalHashTests.cs" />
     <Compile Include="MD5Tests.cs" />
+    <Compile Include="RandomNumberGeneratorTests.netcoreapp.cs" />
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj]: If this is being off in a separate file, shouldn't it be conditionally included?

Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj]: Oops! Yes, I must not have saved before print this up.

File: src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs]: I've been using the ctor. I'm not sure there's a predominant/preferred style yet.

**Comment by stephentoub** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build please ("System.IO.IOException : RemoteInvoke cannot open the remote service. Open Service Status: AppUnavailable" in data tests), cc: @tarekgh

File: src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs [hunk 3]
```diff
@@ -49,6 +51,20 @@ public virtual void GetBytes(byte[] data, int offset, int count)
             }
         }
 
+        public virtual void GetBytes(Span<byte> data)
+        {
+            byte[] array = ArrayPool<byte>.Shared.Rent(data.Length);
+            try
+            {
+                GetBytes(array, 0, data.Length);
+                new ReadOnlySpan<byte>(array, 0, data.Length).CopyTo(data);
```
Comment (davidfowl) [inline src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs]: I guess this is expected to throw if the length is wrong.

**Comment by tarekgh** [comment]
> @dotnet-bot test UWP CoreCLR x64 Debug Build please ("System.IO.IOException : RemoteInvoke cannot open the remote service. Open Service Status: AppUnavailable" in data tests), cc: @tarekgh

This can happen if there is something wrong in the runner app manifest which I doubt that as this failure would show up everywhere. or there is something wrong in the running environment. let me know if you still see the failure and I'll try to look at it.

**Comment by stephentoub** [comment]
> Since RNGCryptoServiceProvider (System.Security.Cryptography.Csp) is public now (ns20)

Thanks for the reminder.  Fixed.

**Pull request merged**